PennMobile/src/main/java/com/pennapps/labs/pennmobile/BookGsrFragment.kt

```diff
@@ -77,6 +77,8 @@ class BookGsrFragment : Fragment() {
                                     || email.text.toString().matches("".toRegex())) {
                                 Toast.makeText(activity, "Please fill in all fields before booking",
                                         Toast.LENGTH_LONG).show()
+                            } else if (!email.text.toString().matches("""[\w]+@(seas.|sas.|wharton.|nursing.)?upenn.edu""".toRegex())) {
```

Comment:
Unfortunately this regex expression isn't completely correct because the "." in regex will match any character except line breaks. Thus, the email "amygutmann@whartonyupennnedu" would be considered valid. You have to add a backslash to actually look for a period character like this:
```
[\w]+@(seas\.|sas\.|wharton\.|nursing\.)?upenn\.edu
```
Details on this site - https://regexr.com/

Reply:
Fixed